solidity/ecdsa/contracts/libraries/EcdsaAuthorization.sol

```diff
@@ -15,25 +15,568 @@
+                uint64(block.timestamp) +
+                self.parameters.authorizationDecreaseDelay;
+        }
+    }
+
+    /// @notice Updates status of the operator in the sortition pool. If there
+    ///         was an authorization decrease requested, it is activated by
+    ///         starting the authorization decrease delay.
+    ///         Function reverts if the operator is not known.
+    function updateOperatorStatus(
+        Data storage self,
+        IStaking tokenStaking,
+        SortitionPool sortitionPool,
+        address operator
+    ) internal {
+        address stakingProvider = self.operatorToStakingProvider[operator];
+        require(stakingProvider != address(0), "Unknown operator");
+
+        AuthorizationDecrease storage decrease = self.pendingDecreases[
+            stakingProvider
+        ];
+
+        emit OperatorStatusUpdated(stakingProvider, operator);
+
+        if (sortitionPool.isOperatorInPool(operator)) {
+            uint96 _eligibleStake = eligibleStake(
+                tokenStaking,
+                stakingProvider,
+                decrease.decreasingBy
+            );
+
+            if (_eligibleStake < self.parameters.minimumAuthorization) {
+                sortitionPool.updateOperatorStatus(operator, 0);
+            } else {
+                sortitionPool.updateOperatorStatus(operator, _eligibleStake);
+            }
+        }
+
+        // If there is a pending authorization decrease request, activate it.
+        // At this point, the sortition pool state is up to date so the
+        // authorization decrease delay can start counting.
+        if (decrease.decreasingAt == type(uint64).max) {
+            decrease.decreasingAt =
+                // solhint-disable-next-line not-rely-on-time
+                uint64(block.timestamp) +
+                self.parameters.authorizationDecreaseDelay;
+        }
+    }
+
+    /// @notice Checks if the operator's authorized stake is in sync with
+    ///         operator's weight in the sortition pool.
+    ///         If the operator is not in the sortition pool and their
+    ///         authorized stake is non-zero, function returns false.
+    function isOperatorUpToDate(
+        Data storage self,
+        IStaking tokenStaking,
+        SortitionPool sortitionPool,
+        address operator
+    ) internal view returns (bool) {
+        address stakingProvider = self.operatorToStakingProvider[operator];
+        require(stakingProvider != address(0), "Unknown operator");
+
+        AuthorizationDecrease storage decrease = self.pendingDecreases[
+            stakingProvider
+        ];
+
+        uint96 _eligibleStake = eligibleStake(
+            tokenStaking,
+            stakingProvider,
+            decrease.decreasingBy
+        );
+
+        if (!sortitionPool.isOperatorInPool(operator)) {
+            return _eligibleStake == 0;
+        } else {
+            return sortitionPool.isOperatorUpToDate(operator, _eligibleStake);
```

Comment:
Shouldn't we add here a similar check we added in `updateOperatorStatus` for `_eligibleStake < self.parameters.minimumAuthorization`?

Actually shouldn't `_eligibleStake < self.parameters.minimumAuthorization` be done inside `eligibleStake` function?

Reply:
Not sure if I understand what is the problem. `eligibleStake` can go below the minimum authorization as a result of slashing from ECDSA or other application. If the sortition pool was locked and operator state in the sortition pool could not be updated, `isOperatorUpToDate` will return `false` because `eligibleStake` does no longer reflect what is in the pool. The operator, or anyone else, should call `updateOperatorStatus` for the problematic operator, and the problematic operator will get removed from the sortition pool as a result of `_eligibleStake < self.parameters.minimumAuthorization` check in `updateOperatorStatus` function. Does it make sense? Am I missing something?

Reply:
I'm sorry for not being clear in my comment. In `updateOperatorStatus` if the eligible stake is below minimum authorization we update the sortition pool to `0`.

```sol
            if (_eligibleStake < self.parameters.minimumAuthorization) {
                sortitionPool.updateOperatorStatus(operator, 0);
            } else {
                sortitionPool.updateOperatorStatus(operator, _eligibleStake);
            }
```

Here in `isOperatorUpToDate` if the eligible stake is below minimum authorization we query sortition pool providing the eligible stake, not `0`. This is inconsistent and may be confusing.

Also, let's imagine a situation where an operator is registered in the sortition pool with minimum authorization `100`. Then governance updates the minimum authorization parameter to `110`. In such a case `isOperatorUpToDate` would still return `true` as it doesn't include the minimum authorization parameter in calculations, but the `updateOperatorStatus` function would update the sortition pool by sending `0` as an authorized stake value removing the operator from the pool.

Reply:
What I was wondering is that instead of checking the calculated eligible stake against the minimum authorization, should we include the `if` statement in the `eligibleStake` function and return `0` if it drops below `minimumAuthorization`.

Reply:
Another option is to add an `if` in `isOperatorUpToDate` to check whether `_eligibleStake < minimumAuthorization` and if this is true and operator is in the pool, return `false`. I think it would address the scenario you mentioned in the comment when the governance altered the minimum authorization value.

I think returning 0 from eligible stake when the authorization is non-zero but below the minimum could also be confusing. After all, there is a stake authorized but it's just below the minimum. On the other hand, we can explain it in the documentation.

I think I am fine either way.